### DISCUSSION TITLE
Add support to pip using git digest

### DISCUSSION THREAD

**Discussion by simao-silva** [renovatebot/renovate#15763] [Suggest an Idea]
### What would you like Renovate to be able to do?

Giving that PIPs can be installed using git (e.g. `git+https://git.example.com/MyProject.git@da39a3ee5e6b4b0d3255bfef95601890afd80709egg=pkg`), add the capability to check for new versions using git digests.

### If you have any ideas on how this should be implemented, please tell us here.

Giving the similarity to the case of Docker image digests, maybe it should be something to consider it. 

### Is this a feature you are interested in implementing yourself?

No

**Comment by viceice** [comment]
git sha are something different that docker digest. but should be easy to implement

**Comment by rarkins** [comment]
Possible to reproduce with a public repo?

**Comment by simao-silva** [comment]
This is an example: <https://github.com/simao-silva/ckan-oauth2-test>. Now, version is at `bf07870b4fda9d9028af1e8367a622a57024b9a4` and there is a newer version with sha `e68dd2664229b7563d77b2c8fc869fe57b747c88`.

**Reply by rarkins** [reply]
That's a Dockerfile. We don't have intentions to support every possible format of every package manager within arbitrary files like that, so you need to use the regex manager.

**Reply by simao-silva** [reply]
I've change the requirements to a `requirements.txt` file. Can you provide support this way?

**Reply by rarkins** [reply]
I'm not familiar with that format in `requirements.txt`. If you look at the files in https://github.com/renovatebot/renovate/tree/main/lib/modules/manager/pip_requirements/__fixtures__ the you can see what we support currently.

**Reply by simao-silva** [reply]
My request is whether that format will be supported. From what I've seen is something similar with [requirements-git-packages.txt](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/pip_requirements/__fixtures__/requirements-git-packages.txt) but instead of a version number we use a git sha/digest.

**Reply by rarkins** [reply]
There's probably nothing stopping it being supported if someone submits a PR
